deployer/deployer-services/src/main/java/io/hyscale/deployer/services/handler/PodParentHelper.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright 2019 Pramati Prism, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.hyscale.deployer.services.handler;
+
+import java.util.List;
+
+import io.hyscale.commons.utils.ResourceLabelUtil;
+import io.hyscale.deployer.core.model.DeploymentStatus;
+import io.kubernetes.client.openapi.ApiClient;
+import io.kubernetes.client.openapi.models.V1ObjectMeta;
+
+public interface PodParentHelper<T> {
+
+    public List<DeploymentStatus> getNotRunningStatusList(ApiClient apiClient, String selector, boolean label,
+            String namespace);
+
+    public DeploymentStatus getNotRunnnigStatus(T t);
+
+    public List<DeploymentStatus> getNotRunningStatusList(List<T> t);
```

Comment:
You are just building, right? Also pass the status as an option to build, which makes it more extensible.

Reply:
Applies to getNotRunnnigStatus() also.

getNotRunnnigStatus is not a contract of the class